test_runner/regress/test_pageserver_generations.py

```diff
@@ -276,15 +277,28 @@ def test_deferred_deletion(neon_env_builder: NeonEnvBuilder):
     assert get_deletion_queue_unexpected_errors(ps_http) == 0
 
 
-@pytest.mark.parametrize("keep_attachment", [True, False])
-@pytest.mark.parametrize("validate_before", [True, False])
+class KeepAttachment(str, enum.Enum):
+    KEEP = "keep"
+    LOSE = "lose"
+
+
+class ValidateBefore(str, enum.Enum):
```

Comment:
This just remakes a boolean? I understand when you replace booleans in complicated Rust APIs where there is a lot of hierarchy, it is simpler to have `foo(KEEP, VALIDATE)` than to have `foo(true, true)`. But a) we are in python so you can also just use named arguments instead of positional ones, and b) there is not a lot of hierarchy.

Or is there some change in the way the test is printed when running it? If there is a strong argument for having it, I'm fine with the change. But it needs a good argument :)

Reply:
> Or is there some change in the way the test is printed when running it? 

Right, this is the reason.  Pytest runs in a result report are suffixed with the values of their parametrized arguments (but not their keys).  So when using booleans here you get rather enigmatically named  "testname-True-False" -- the purpose of the enum is to replace those trues/false with something that is meaningful without cross-referencing to the order of boolean parameters.